ReactAndroid/src/main/java/com/facebook/react/XReactInstanceManagerImpl.java

```diff
@@ -811,7 +812,7 @@ private void detachViewFromInstance(
       CatalystInstance catalystInstance) {
     UiThreadUtil.assertOnUiThread();
     catalystInstance.getJSModule(AppRegistry.class)
-        .unmountApplicationComponentAtRootTag(rootView.getId());
```

Comment:
It's one of the few places where it's needed with this approach mainly because it's calling JS which isn't aware of the mapping.